server/src/main/java/org/opensearch/wlm/stats/QueryGroupStats.java

```diff
@@ -0,0 +1,197 @@
+        private final Map<ResourceType, ResourceStats> resourceStats;
+
+        public QueryGroupStatsHolder(long completions, long rejections, Map<ResourceType, ResourceStats> resourceStats) {
+            this.completions = completions;
+            this.rejections = rejections;
+            this.resourceStats = resourceStats;
+        }
+
+        public QueryGroupStatsHolder(StreamInput in) throws IOException {
+            this.completions = in.readVLong();
+            this.rejections = in.readVLong();
+            this.resourceStats = in.readMap((i) -> ResourceType.fromName(i.readString()), ResourceStats::new);
+        }
+
+        /**
+         * Writes the {@param statsHolder} to {@param out}
+         * @param out StreamOutput
+         * @param statsHolder QueryGroupStatsHolder
+         * @throws IOException exception
+         */
+        public static void writeTo(StreamOutput out, QueryGroupStatsHolder statsHolder) throws IOException {
+            out.writeVLong(statsHolder.completions);
+            out.writeVLong(statsHolder.rejections);
+            out.writeMap(statsHolder.resourceStats, (o, val) -> o.writeString(val.getName()), ResourceStats::writeTo);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            QueryGroupStatsHolder.writeTo(out, this);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.field(COMPLETIONS, completions);
+            builder.field(REJECTIONS, rejections);
+            for (Map.Entry<ResourceType, ResourceStats> resourceStat : resourceStats.entrySet()) {
+                ResourceType resourceType = resourceStat.getKey();
+                ResourceStats resourceStats1 = resourceStat.getValue();
+                builder.startObject(resourceType.getName());
+                resourceStats1.toXContent(builder, params);
+                builder.endObject();
+            }
+            return builder;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            QueryGroupStatsHolder that = (QueryGroupStatsHolder) o;
+            return completions == that.completions && rejections == that.rejections && Objects.equals(resourceStats, that.resourceStats);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(completions, rejections, resourceStats);
+        }
+    }
+
+    /**
+     * point in time resource level stats holder
+     */
+    public static class ResourceStats implements ToXContentObject, Writeable {
+        public static final String CURRENT_USAGE = "current_usage";
+        public static final String CANCELLATIONS = "cancellations";
+        private final double currentUsage;
+        private final long cancellations;
```

Comment:
I think it would provide more visibility and help us correlate when we need to tie up rejections metric with high resource metric. I will add this at resource level.